### DISCUSSION TITLE
We need your help: More tutorials and example applications for the docs!

### DISCUSSION THREAD

**Discussion by fwitte** [oemof/tespy#506] [Ideas]
Hi all,

a large variety of discussions has been going on recently, which I am really happy about. Based on this, I thought, that I would like to see more examples/tutorials to the documentation page of TESPy to reflect the various possibilities TESPy offers. To do that, I would like to reach out to all users.

## How to contribute

**Comment in this thread in less than 50 words:**

- What your model is about (type of plant, and what investigation to do on it).
- What the libraries are you are using in the model apart from tespy.

As a rule of thumb, I aim to include examples with the following specifics

- There is a short "storyline" for the model
  - What do you want to analyze?
  - What can we learn from the model?
  - What are the challenges in the model implementation?
  - How can we extract and visualize results?
- Small to mid-size models (anything with roughly 50 connections or less)

## Modeling context

Highly welcome examples would cover the following topics

- ORC power plants
- transcritical and/or supercritical heat pump applications
- gas turbine power plants
- waste heat recovery applications
- flash processes
- model coupling with any kind of storage
- cooling machines/air conditioning or cryogenic applications
- any kind of industrial processing e.g. of food or beverages
- usage of custom fluid property functions
- usage of custom components
- storage integration in tespy based simulation

## Procedure

1. I would like to collect possible examples in this discussion thread until the online user meeting on June 17.
2. Then I would like to go over the examples in the June user meeting and collect suitable ones.
3. Once we have those, I would like to draft some kind of loose template structure (which would be similar to those already on the documentation page). The examples should then be transferred to the template.

Looking forward to the contributions!

Have a sunny weekend, all the best

Francesco

**Comment by maltefritz** [comment]
Hi @fwitte,

@jfreissmann and I have developed an open-source web dashboard for simulating, analyzing and evaluating heat pumps based on TESPy. 

### Heat Pump Simulator
With this dashboard, a variety of complex thermodynamic system models can be controlled using numerical methods via a simple interface. This includes both the design of heat pumps and the simulation of their stationary partial load operation. The results of the simulations provide information about the basic behavior, the COP, state variables, results of exergy analysis and costs of the individual components as well as the total investment costs of the heat pump under consideration. This provides access to complex issues that regularly arise in the design and planning of heat pumps.

### Key Features
  - Steady-state simulation of design and partload operation based on [TESPy](https://github.com/oemof/tespy)
  - Parametrization and result visualisation through a [Streamlit](https://github.com/streamlit/streamlit) dashboard
  - Industry standard, as well as topologies still in research and developement, supported
  - Sub- and transcritical processes
  - Wide variety of refrigerants due to the integration of [CoolProp](https://github.com/CoolProp/CoolProp)
  - Contains various states-diagrams using [fluprodia ](https://github.com/fwitte/fluprodia)

It can be found at [heatpumps](https://github.com/jfreissmann/heatpumps) and everyone is welcome to try out and further develop the tool. I think this could a great example for documentation.

**Reply by fwitte** [reply]
Thank you for your suggestion! :)

**Comment by maltefritz** [comment]
This tool could also be used to create a tutorial for a transcritical heat pump application. If you like, a manual could be created to implement this step by step.

Best regards

**Comment by PabloMBarral** [comment]
Tespy is suitable to perform CHP cycles. Steam turbine with extraction, bfw tank, hrsgs. With three or four models a quick feasibility study can be performed.

With that, marginal efficiencies, thermal and exergetic efficiencies, and savings.

I'll attach some drawings.

**Reply by PabloMBarral** [reply]
Tespy is suitable for solving Rankine cycles and using exergetic cost theory.

And, in a lower level, can be used to solve cycles of lp steam in boiler room. I mean, flashing blowdown, adding preheaters, etc.

**Reply by PabloMBarral** [reply]
![lp cycle - III.png](https://github.com/oemof/tespy/assets/45795487/67310061-7f94-4bb6-9080-2699a42a117a)

![hp cycle - III.png](https://github.com/oemof/tespy/assets/45795487/0de4ed83-18e2-4207-9a42-039e6f5b4d1e)

I mean this cycles. I have done the lp cycle, it's in this forum.

I also have data of real hrsgs, PFDs and so. That could be a great example.

I recall some paper with a comprehensive model of a gas turbine. Sorgenfrei & Tsatsaronis 2016. 

I use ees for everything, but tespy is more straightforward. I even wonder if the coding can be chatgpted.

**Reply by PabloMBarral** [reply]
Last message, I'm on my cell and it's late.

CHP, both lp and hp cycles, can show how exergy is destroyed, and how this destruction can be reduced.

Lp cycles are extensively used in industry with little exergetic concern. Everyone thinks that are great because its excellent first law efficiency. 

CHP feasibility analysis does not use exergy directly, but it is implicit in the high marginal efficiency. 

And with the pre established models one can provide proposal engineers with some models to calculate savings without need to dig in thermal balances. Perhaps integration with excel could be used. 

Regards!

**Reply by fwitte** [reply]
Hi @PabloMBarral,

thank you for the suggestion!

Have a nice weekend, best

Francesco

**Reply by PabloMBarral** [reply]
Let me know if you need some help. Regards!

**Comment by jgunstone** [comment]
# Main Plant with Thermal Energy Storage

following up from here: https://github.com/oemof/tespy/issues/423
@gregorlein has imlemented something along these lines here: https://github.com/oemof/tespy/discussions/505

For modelling the main heating plant for large-scale: commercial, public or multi-residential buildings. 
Thermal Energy storage is essential in such systems to reduce the peak load to met by the plant, balance the supply to the variable demand, optimise the efficiency of the heat-pump when running by operating at a higher, more constant load. 

The demand (associated with the building) would be calculated in a different software (energyplus/honeybee, IES, or a maybe oemf-demandlib (?)). Typically annual hourly, but could begin with an example day. 

The analysis should help yield: 
- the min size of heat plant and the required size of thermal store to meet the peak demand

Starting from the DHN example and extending to include storage: 

![image](https://github.com/oemof/tespy/assets/21370980/65b0f7fe-1a56-4e67-877e-0ba57aa04c60)



It would then be great to make the heating plant a Heat Pump 
i.e. 
![image](https://github.com/oemof/tespy/assets/21370980/fbba9569-0a7a-44b8-8a24-05cfc549c0f6)
then could calc CoP of the heating plant based on part-load efficiencies
perhaps could link to / integrate with https://github.com/oemof/tespy/discussions/506#discussioncomment-9329598 ... 

(excuse the scruffy sketches :) )

**Reply by fwitte** [reply]
@jgunstone, sorry for the kind of late reply. Excellent idea, thank you for suggesting it :)

**Reply by gregorlein** [reply]
Hi!
yes, I implemented a daily buffer in Tespy with the user defined equations to shave the peak demand and reduce the size of the heat pump in my system. I think it works, but I didn't continue on this topic as it wasn't that relevant for my project in the end... ;)

However, I still do use TESPy for analizing a seasonal storage. My project is on an Aquifer Thermal Energy Storage (ATES). I have coupled a TESPy model that simulates flows and temperatures on the building level, including a heat pump, to a numerical model that simulates the temperature changes in the aquifer. Basically TESPy computes the flows to discharge/charge the ATES and the numerical model updates the temperatures for a heat source in the TESPy model.

The model is probably too big for a demonstration. But maybe I could contribute to a storage tutorial. 
For me, TESPy was most helpful for comparing different design options, so where to connect the heat pump and storage in the system (e.g. compare COP, flow rates, temperatures...)
